learn/guides/asyncapi-tool.md

```diff
@@ -16,144 +16,147 @@ The Ballerina AsyncAPI tool will make it easy for you to start the development o
 
-Execute the command below on the above file. This will generate four files as mentioned above.
+  > **Note:** Currently, this tool supports only HTTP-based event APIs.
 
-```bash
-bal asyncapi -i slack.yaml
-```
+2\. `x-ballerina-event-type` - This should be there in every event inside the channel. This is the name of the event or the value of the above mentioned attribute for a specific event.
 
-### Modify the generated sources
+## Generate Ballerina services from AsyncAPI Contracts
 
-The generated files can be modified according to the custom requirements. When modifying the generated code segments, it will be easier to consider the facts below.
+After modifying the AsyncAPI contract, the Ballerina sources can be generated using the below commands.
 
-- All the incoming requests will be coming to the resource function in the `dispatcher_service.bal` file. Hence, if there is a necessity to add an authentication logic for the incoming calls, that logic can be included there before processing the incoming HTTP request.
-- If more information is needed when initializing the listener like secrets, endpoint URLs, tokens, refresh tokens, etc., update the `init` function in the `listener.bal` file.
+```bash
+bal asyncapi [-i | --input] <asyncapi-contract-file-path> [-o | --output] <output-location>
+```
 
-### Execute the generated sources
+The generated service can be used as a code template to start the service implementation.
+For example,
 
-Follow the steps below to use the generated Ballerina sources.
+```bash
+bal asyncapi -i hello.yaml
+```
+
+This generates a Ballerina source (i.e., the four Ballerina files below) from the given AsyncAPI definition file.
 
-1\. Create an empty folder and copy all the generated files to it.
+1. `data_types.bal` - contains all the Ballerina data types extracted from the AsyncAPI definition
+2. `service_types.bal` - contains all the service types relevant to the event API described in the AsyncAPI definition
+3. `listener.bal` - contains the HTTP listener, which listens to the relevant third-party service
+4. `dispacther_service.bal` - contains the event dispatching logic
 
-2\. Execute the command below from the root of the folder. This generates a `Ballerina.toml` file in the root of the folder.
+The generated Ballerina sources will be written into the same directory from which the command is run. The above command can be run from anywhere on the execution path. It is not mandatory to run it from within a Ballerina package. If you want to generate Ballerina sources to a specific provided output location, you can modify the above command as below.
 
 ```bash
-bal init
+bal asyncapi -i hello.yaml -o ./output_path
 ```
 
-3\. Create a new Ballerina file inside the folder (e.g., `slack_service.bal`) and paste the code below to it.
+Then, the generated files can be modified according to the custom requirements. When modifying the generated code segments, it will be easier to consider the below facts
```

Comment:
```suggestion
Then, the generated files can be modified according to the custom requirements. When modifying the generated code segments, it will be easier to consider the below facts.
```
```suggestion
Then, the generated files can be modified according to the custom requirements. When modifying the generated code segments, it will be easier to consider the below facts
```